keel-schema-generator/src/main/kotlin/com/netflix/spinnaker/keel/schema/Schema.kt

```diff
@@ -79,21 +84,29 @@ data class Reference(
 
 data class OneOf(
   override val description: String?,
-  val oneOf: Set<Schema>,
-  val discriminator: Discriminator? = null
-) : Schema {
-  data class Discriminator(
-    val propertyName: String,
-    val mapping: SortedMap<String, String>
-  )
-}
+  val oneOf: Set<Schema>
+) : Schema
 
 data class AllOf(
   val allOf: List<Schema>
 ) : Schema {
   override val description: String? = null
 }
 
+data class ConditionalSubschema(
+  val `if`: Condition,
+  val then: Subschema
+)
+
+data class Condition(
+  val properties: Map<String, ConstSchema>
+)
+
+data class Subschema(
+  val properties: Map<String, Schema>,
+  val required: SortedSet<String> = emptySet<String>().toSortedSet()
+)
+
 /**
  * Yes, I really had to implement an either monad to get this all to work.
```

Comment:
Never apologize for implementing monads.

Reply:
I'm kind of amazed Kotlin doesn't have one by default